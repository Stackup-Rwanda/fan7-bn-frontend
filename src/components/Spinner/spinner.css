.loading {
  width: 100px;
  height: 100px;
  margin: 22% auto;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  border-radius: 50%;
  border-top: 5px solid #A2AAB1;
  position: relative;
  vertical-align: middle;
  -webkit-animation: a1 2s linear infinite;
          animation: a1 2s linear infinite;
}

.loading::before, .loading::after {
  content: '';
  width: 100px;
  height: 100px;
  position: absolute;
  left: 0;
  top: -5px;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  border-top: 5px solid #A2AAB1;
  border-radius: 50%;
}

.loading::before {
  -webkit-transform: rotate(120deg);
          transform: rotate(120deg);
}

.loading::after {
  -webkit-transform: rotate(240deg);
          transform: rotate(240deg);
}

.loading span {
  position: absolute;
  width: 100px;
  height: 100px;
  color: #000000;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  text-align: center;
  line-height: 100px;
  -webkit-animation: a2 2s linear infinite;
          animation: a2 2s linear infinite;
}

@-webkit-keyframes a1 {
  to {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}

@keyframes a1 {
  to {
    -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
  }
}

@-webkit-keyframes a2 {
  to {
    -webkit-transform: rotate(-360deg);
            transform: rotate(-360deg);
  }
}

@keyframes a2 {
  to {
    -webkit-transform: rotate(-360deg);
            transform: rotate(-360deg);
  }
}
/*# sourceMappingURL=spinner.css.map */