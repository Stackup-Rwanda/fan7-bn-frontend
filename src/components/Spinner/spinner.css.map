{
    "version": 3,
    "mappings": "AAAA,AAAA,QAAQ,CAAC;EACL,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,MAAM,EAAE,QAAQ;EAChB,UAAU,EAAE,UAAU;EACtB,aAAa,EAAE,GAAG;EAClB,UAAU,EAAE,iBAAiB;EAC7B,QAAQ,EAAE,QAAQ;EAClB,cAAc,EAAE,MAAM;EACtB,SAAS,EAAE,qBAAqB;CAiCnC;;AA1CD,AAWI,QAXI,AAWH,QAAQ,EAXb,QAAQ,AAYH,OAAO,CAAC;EACL,OAAO,EAAE,EAAE;EACX,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,QAAQ,EAAE,QAAQ;EAClB,IAAI,EAAE,CAAC;EACP,GAAG,EAAE,IAAI;EACT,UAAU,EAAE,UAAU;EACtB,UAAU,EAAE,iBAAiB;EAC7B,aAAa,EAAE,GAAG;CACrB;;AAtBL,AAwBI,QAxBI,AAwBH,QAAQ,CAAC;EACN,SAAS,EAAE,cAAc;CAC5B;;AA1BL,AA4BI,QA5BI,AA4BH,OAAO,CAAC;EACL,SAAS,EAAE,cAAc;CAC5B;;AA9BL,AAgCI,QAhCI,CAgCJ,IAAI,CAAC;EACD,QAAQ,EAAE,QAAQ;EAClB,KAAK,EAAE,KAAK;EACZ,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,OAAO;EACd,UAAU,EAAE,UAAU;EACtB,UAAU,EAAE,MAAM;EAClB,WAAW,EAAE,KAAK;EAClB,SAAS,EAAE,qBAAqB;CACnC;;AAGL,UAAU,CAAV,EAAU;EACN,EAAE;IACE,SAAS,EAAE,cAAc;;;;AAIjC,UAAU,CAAV,EAAU;EACN,EAAE;IACE,SAAS,EAAE,eAAe",
    "sources": [
        "spinner.scss"
    ],
    "names": [],
    "file": "spinner.css"
}