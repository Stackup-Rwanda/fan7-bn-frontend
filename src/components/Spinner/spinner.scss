.loading {
    width: 100px;
    height: 100px;
    margin: 22% auto;
    box-sizing: border-box;
    border-radius: 50%;
    border-top: 5px solid #A2AAB1;
    position: relative;
    vertical-align: middle;
    animation: a1 2s linear infinite;

    &::before, 
    &::after {
        content: '';
        width: 100px;
        height: 100px;
        position: absolute;
        left: 0;
        top: -5px;
        box-sizing: border-box;
        border-top: 5px solid #A2AAB1;
        border-radius: 50%;
    }

    &::before {
        transform: rotate(120deg);
    }

    &::after {
        transform: rotate(240deg);
    }

    span {
        position: absolute;
        width: 100px;
        height: 100px;
        color: #000000;
        box-sizing: border-box;
        text-align: center;
        line-height: 100px;
        animation: a2 2s linear infinite;
    }
}

@keyframes a1 {
    to {
        transform: rotate(360deg);
    }
}

@keyframes a2 {
    to {
        transform: rotate(-360deg);
    }
}