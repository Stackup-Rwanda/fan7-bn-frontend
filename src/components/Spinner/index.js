import React from 'react';
import './spinner.scss';

const Loading = () => (
  <div className="loading">
    <span>Loading...</span>
  </div>
);

export default Loading;
