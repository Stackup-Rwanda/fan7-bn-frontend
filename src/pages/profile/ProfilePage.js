import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Card from '../../components/profile/ProfileCard';
import ProfileFeeds from '../../components/profile/ProfileFeeds';
import ProfileTrips from '../../components/profile/ProfileTrips';
import DashboardLayout from '../../components/DashboardLayout';
import Spinner from '../../components/Spinner';
import profileActions from '../../store/modules/profile/actions';
import selectors from '../../store/modules/profile/selectors';
import './styles/profile.scss';

class ProfilePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: 'activityFeeds',
    };

    this.handleNavigation = this.handleNavigation.bind(this);
  }

  componentDidMount() {
    const { dispatch } = this.props;
    const { getProfile, getTrips } = profileActions;
    dispatch(getProfile());
    dispatch(getTrips());
  }

  handleNavigation(value) {
    this.setState({ show: value });
  }

  render() {
    const { show } = this.state;
    const {
      fullName, userName, gender, dob, address, profileImg, trips, loading,
    } = this.props;

    return loading ? (
      <Spinner />
    ) : (
      <div className="profile">
        <div className="profile-hero">
          <h1 className="profile-hero__title">Travel The World</h1>
        </div>
        <div className="profile-card">
          <Card
            fullName={fullName}
            userName={userName}
            profileImg={profileImg}
            handleNavigation={this.handleNavigation}
          />
        </div>
        <div className="profile-content">
          {show === 'activityFeeds' && <ProfileFeeds gender={gender} dob={dob} address={address} />}
          {show === 'trips' && <ProfileTrips trips={trips} />}
        </div>
      </div>
    );
  }
}

ProfilePage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  fullName: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  dob: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  profileImg: PropTypes.string.isRequired,
  trips: PropTypes.arrayOf().isRequired,
};

const {
  selectLoading,
  selectProfileFullName,
  selectProfileUserName,
  selectProfileGender,
  selectProfileDob,
  selectProfileAddress,
  selectProfileAvatar,
  selectProfileRole,
} = selectors;

const select = ({ userProfile }) => ({
  loading: selectLoading(userProfile),
  fullName: selectProfileFullName(userProfile),
  userName: selectProfileUserName(userProfile),
  gender: selectProfileGender(userProfile),
  dob: selectProfileDob(userProfile),
  address: selectProfileAddress(userProfile),
  profileImg: selectProfileAvatar(userProfile),
  role: selectProfileRole(userProfile),
  trips: userProfile ? userProfile.trips : null,
});

export default connect(select)(DashboardLayout(ProfilePage));
