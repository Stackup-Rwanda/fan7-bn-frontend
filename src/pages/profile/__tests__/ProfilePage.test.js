/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfilePage from '../ProfilePage';
import Spinner from '../../../components/Spinner';
import Card from '../../../components/profile/ProfileCard';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const history = {
  push: jest.fn(),
  replace: jest.fn(),
  createHref: jest.fn(),
};

describe('ProfilePage Component Test', () => {
  it('should return true if component exists', () => {
    const initialState = {};
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ProfilePage />
        </Router>
      </Provider>,
    );

    expect(wrapper.find('.profile').exists()).toBe(true);
  });

  it('should display spinner if it\'s loading', () => {
    const initialState = {
      userProfile: { loading: true },
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ProfilePage />
        </Router>
      </Provider>,
    );

    expect(wrapper.find(Spinner).exists()).toBe(true);
  });

  it('should display content if it is not loading', () => {
    const initialState = {
      userProfile: { loading: false },
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ProfilePage />
        </Router>
      </Provider>,
    );

    expect(wrapper.find('.profile-card').exists()).toBe(true);
    expect(wrapper.find('.profile-content').exists()).toBe(true);

    wrapper
      .find(Card)
      .props()
      .handleNavigation('trips');
  });
});

describe('<ProfilePage /> shallow rendering tests', () => {
  const initialState = {};
  const store = mockStore(initialState);

  it('matches the snapshot', () => {
    const tree = shallow(<ProfilePage store={store} />);
    expect(tree).toMatchSnapshot();
  });
});
