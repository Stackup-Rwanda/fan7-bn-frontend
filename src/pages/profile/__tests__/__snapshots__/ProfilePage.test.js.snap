// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> shallow rendering tests matches the snapshot 1`] = `
<ContextProvider
  value={null}
>
  <_class
    address=""
    dispatch={[Function]}
    dob=""
    fullName="Provide your names"
    gender=""
    loading={false}
    profileImg=""
    role=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    trips={null}
    userName=""
  />
</ContextProvider>
`;
